db/write_batch.cc

```diff
@@ -2500,28 +2500,48 @@ class MemTableInserter : public WriteBatch::Handler {
       if (cf_handle == nullptr) {
         cf_handle = db_->DefaultColumnFamily();
       }
-      Status get_status = db_->Get(read_options, cf_handle, key, &get_value);
+
+      Status get_status =
+          db_->GetEntity(read_options, cf_handle, key, &existing);
       if (!get_status.ok()) {
         // Failed to read a key we know exists. Store the delta in memtable.
         perform_merge = false;
       } else {
-        Slice get_value_slice = Slice(get_value);
-
         // 2) Apply this merge
         auto merge_operator = moptions->merge_operator;
         assert(merge_operator);
 
+        const auto& columns = existing.columns();
+
+        Status merge_status;
         std::string new_value;
         ValueType new_value_type;
-        // `op_failure_scope` (an output parameter) is not provided (set to
-        // nullptr) since a failure must be propagated regardless of its value.
-        Status merge_status = MergeHelper::TimedFullMerge(
-            merge_operator, key, MergeHelper::kPlainBaseValue, get_value_slice,
-            {value}, moptions->info_log, moptions->statistics,
-            SystemClock::Default().get(),
-            /* update_num_ops_stats */ false, &new_value,
-            /* result_operand */ nullptr, &new_value_type,
-            /* op_failure_scope */ nullptr);
+
+        if (columns.size() == 1 &&
+            WideColumnsHelper::HasDefaultColumn(columns)) {
```

Comment:
Unrelated to this PR, but a question as a c++ n00b. Would `inline`ing this function in `WideColumnsHelper` would help the performance?

Reply:
Well, `HasDefaultColumn` is already implicitly inlined in `WideColumnsHelper`, since it's defined (as opposed to just declared) inside the definition of `WideColumnsHelper`. As for the performance effect of inlining, there are no general answers here; it can help performance in cases where the function is short and performance critical but sometimes it actually hurts performance by increasing code bloat. To complicate matters further, `inline` is for the most part just a hint to the compiler, which might or might not honor it 😀